src/main/kotlin/car/racing/CarRacing.kt

```diff
@@ -0,0 +1,54 @@
+package car.racing
+
+import kotlin.random.Random
+
+class Car(
+    private val forwardThreshold: Int = CAN_GO_FORWARD_THRESHOLD
+) {
+    private val random = Random(System.currentTimeMillis())
+
+    private var _moves: Int = 0
+    val moves: Int
+        get() = _moves
+
+    fun run() {
+        if (canGoForward()) {
+            _moves += 1
+        }
+    }
+
+    private fun canGoForward(): Boolean {
+        require(forwardThreshold in CAN_GO_FORWARD_MIN..CAN_GO_FORWARD_MAX)
+        return random.nextInt(CAN_GO_FORWARD_MIN, CAN_GO_FORWARD_MAX) >= forwardThreshold
```

Comment:
```suggestion
        return (CAN_GO_FORWARD_MIN..CAN_GO_FORWARD_MAX).random() >= forwardThreshold
```
별도의 random 객체 생성 없이 kotlin range의 random() 함수를 사용하시면 이렇게도 구현할 수 있습니다! kotlin에는 생각지도 못한 유용한 extension function이 많이 존재합니다 🙂

Reply:
그렇군요.🤔
`(CAN_GO_FORWARD_MIN..CAN_GO_FORWARD_MAX).random()` 에서 매번 IntRange를 생성하게 되는 문제가 있어보이네요. 
Random 기본 객체를 사용하는거는 어떤가요? `Random.nextInt(0, 9)`

Reply:
매번 IntRange를 생성하는 문제를 해결하려면 다음과 같이 상수로 분리하는 방법이 있을 것 같아요! 
```kotlin
    companion object {
        private const val CAN_GO_FORWARD_MIN = 0
        private const val CAN_GO_FORWARD_MAX = 9
        private const val CAN_GO_FORWARD_THRESHOLD = 4
        private val CAN_GO_FORWARD_RANGE = CAN_GO_FORWARD_MIN..CAN_GO_FORWARD_MAX
    }
```
제안 주신 Random 기본 객체를 사용하는 것도 별도의 random 객체 생성 없이 처리할 수 있는 좋은 방법이라고 생각합니다 👍